### ISSUE TITLE
Segmentation Fault

### ISSUE THREAD

**Issue by adamstambouli** [blenderskool/pigmnts#46]
Hey! 

I'm getting a segmentation fault on a M1 Macbook Pro running Sonoma 14.0.

```sh
pigmnts test.png -xrd
Creating a palette of 5 colors from test.png
⠸
[1]    16785 segmentation fault  pigmnts test.png -xrd
```

Any logs I can check to see how to fix this?

**Comment by blenderskool** [comment]
@adamstambouli Are you using one of the binaries from releases?
There is no binary for M1 chips([Ref](https://github.com/blenderskool/pigmnts/blob/ba7ceb4b7bef864dff14d396b4c5ec6fd40f2e49/.github/workflows/cli.yml#L28-L30)), so either you'll have to compile from source, or we can update the build CI with M1 as another target. Would you be interested in contributing this change?

**Comment by adamstambouli** [comment]
@blenderskool oh gotcha! Yes I'm happy to contribute although I'm not familiar with Rust. 

Do you have any advice/tips on which changes I should be making?

**Comment by blenderskool** [comment]
I don't think it should require any rust code changes. Try the following steps:
1. Build the binary on your local M1 machine by cloning the project and running `cargo build --release`. This requires rust to be installed locally. If the binary works perfectly, we can add it in our CI config.
2. Just add another entry for macOS on arm here. We'll need `macos-14` from this list [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) to build for M1 chips. https://github.com/blenderskool/pigmnts/blob/ba7ceb4b7bef864dff14d396b4c5ec6fd40f2e49/.github/workflows/cli.yml#L21-L30

**Comment by adamstambouli** [comment]
So the good news is that building the binary runs successfully on Apple Silicon. I used:

```sh
cargo build --release --target aarch64-apple-darwin
```

Bad news is that the segfault still there:

```sh
./target/aarch64-apple-darwin/release/pigmnts ~/Downloads/test.png
```

```sh
Creating a palette of 5 colors from /Users/xxx/Downloads/test.png
⠹
[1]    20568 segmentation fault  ./target/aarch64-apple-darwin/release/pigmnts ~/Downloads/test.png
```

It's possible this is an Apple Silicon issue since Rust doesn't fully support it as a target: https://github.com/rust-lang/rust/issues/73908

**Comment by blenderskool** [comment]
Hmm, could you share the image file you are using? I'll test it on my amd64 machine to see if it's an issue with our code.

**Comment by adamstambouli** [comment]
Yeah no problem, here it is.

[pigmnts.zip](https://github.com/blenderskool/pigmnts/files/14396732/pigmnts.zip)

**Comment by blenderskool** [comment]
@adamstambouli Sorry, I was asking for the the image file you were using. I'll see if there's some edge case that went unhandled with the image you had used.

**Comment by adamstambouli** [comment]
oh 😆 here are a few sample images that cause segfaults: [pigmnts-sample-images.zip](https://github.com/blenderskool/pigmnts/files/14397399/pigmnts-sample-images.zip)

```sh
pigmnts arsenal-away-23-24.webp
Creating a palette of 5 colors from arsenal-away-23-24.webp
⠹
[1]    38020 segmentation fault  pigmnts arsenal-away-23-24.webp
```

```sh
pigmnts united-third-23-24.png
Creating a palette of 5 colors from united-third-23-24.png
⠹
[1]    38063 segmentation fault  pigmnts united-third-23-24.png
```

```sh
pigmnts en.wikipedia.org_wiki_Liverpool_F.C..png
Creating a palette of 5 colors from en.wikipedia.org_wiki_Liverpool_F.C..png
⠋
[1]    38250 segmentation fault  pigmnts en.wikipedia.org_wiki_Liverpool_F.C..png
```

**Comment by blenderskool** [comment]
@adamstambouli Seems like this is an M1 specific issue since all the images worked fine on my x64 machine. Yet to build on M1 and test it.

**Comment by adamstambouli** [comment]
@blenderskool thanks for investigating! Please feel free to triage this issue as you wish - it's ok if you close it and add M1 support to the roadmap if others are interested in the functionality.
